nilearn/surface/tests/test_surface.py

```diff
@@ -364,17 +391,20 @@ def test_load_surf_mesh_file_glob():
     assert len(load_surf_mesh(fname1)) == 2
     assert_array_almost_equal(load_surf_mesh(fname1)[0], mesh[0])
     assert_array_almost_equal(load_surf_mesh(fname1)[1], mesh[1])
-
     os.remove(fname1)
     os.remove(fname2)
```

Comment:
> I'm not sure to understand why we wouldn't want to remove them. I'm afraid this could potentially cause some failures. For example if a test creates some of these files and later does something different using wildcards. It would then grab unexpectedly left over garbage files. Does this make sense?

the issue is removing them in statements that may never be reached (as opposed
to removing them in `__exit__` methods or `finally` blocks). if we look at this
one for example,

- if removing the temporary file is necessary, there is a bug here. if control
  leaves the function somewhere else due to an exception or a return statement,
  the file is never removed. that was the case before you changed this function
  to use `tmp_path`.
- if removing the file is not necessary, removing it makes it less obvious that
  in any case it is actually being dealt with somewhere else (by the pytest
  fixture, in this case). this is the current situation now that you fixed the
  bug. note that many other tests in the library create files in `tmp_path` and
  never remove them -- dealing with them is pytest's job -- and if we remove all
  the `os.remove` calls in this file the tests still work.

but it's also not an error to remove them so it's not that important; if you
would rather remove them let's keep it like this! and it's not really related with
replacing `mktemp` with `mkstemp` anyway. (btw now that the files are created in
an empty scratch directory the need for `mkstemp` is smaller)

(in a similar way there are some `del` in this file for which the need is not clear)